serpent-moltres/fuel-column-sensitivity/postprocessing.py

```diff
@@ -0,0 +1,518 @@
+        r = np.sqrt(x1**2 + y1**2)
+        x = r
+    else:
+        x = file1[dire].tolist()
+
+    x = np.array(x)
+    group11 = file1['group1'].tolist()
+    N = len(group11)
+    # the number of columns in the csv file is the number of
+    # groups + 4: x, y, z, id.
+    G = len(file1.keys()) - 4
+
+    Gp = len(lim1)
+    group1 = np.zeros((Gp, N))
+
+    for gp in range(Gp):
+        if gp == 0:
+            for g in range(lim1[0]):
+                group1[gp] += np.array(file1['group'+str(g+1)].tolist())
+        else:
+            for g in range(lim1[gp-1], lim1[gp]):
+                group1[gp] += np.array(file1['group'+str(g+1)].tolist())
+
+    group21 = file2['group1'].tolist()
+    N = len(group21)
+    # the number of columns in the csv file is the number of
+    # groups + 4: x, y, z, id.
+    G = len(file2.keys())-4
+
+    Gp = len(lim2)
+    group2 = np.zeros((Gp, N))
+
+    for gp in range(Gp):
+        if gp == 0:
+            for g in range(lim2[0]):
+                group2[gp] += np.array(file2['group'+str(g+1)].tolist())
+        else:
+            for g in range(lim2[gp-1], lim2[gp]):
+                group2[gp] += np.array(file2['group'+str(g+1)].tolist())
+
+    e = np.zeros(3)
+    for gp in range(Gp):
+        aux = ((group1[gp]-group2[gp])/group1[gp])**2
+        e[gp] = np.sqrt(sum(aux))
+
+    # active core error
+    group1 = np.where(x >= 160, group1, 0)
+    group1 = np.where(x <= (160 + 793), group1, 0)
+    group2 = np.where(x >= 160, group2, 0)
+    group2 = np.where(x <= (160 + 793), group2, 0)
+
+    e2 = np.zeros(3)
+    for gp in range(Gp):
+        aux = group1[gp][group1[gp] != 0]-group2[gp][group2[gp] != 0]
+        aux /= group1[gp][group1[gp] != 0]
+        aux = aux**2
+        e2[gp] = np.sqrt(sum(aux))
+
+    return e, e2
+
+
+def plot_error_acc_study(error, save, xticks, xlabel):
+    '''
+    This function plots the error for different energy group structures.
+    In this case, the energy group structures are labeled a, b, c, d, e.
+
+    Parameters:
+    -----------
+    error: [ 5 x 2 x 3 array]
+        error for the different energy group structures (5)
+        error in the whole domain and in the active core region (2)
+        error for the 3 fluxes (3)
+    save: [string]
+        name of the figure to produce
+    xticks: [list]
+        list with x values for the plot
+    xlabel: [string]
+        xlabel text
+    '''
+
+    plt.figure()
+
+    plt.plot(xticks, error[:, 1, 0], marker='o', label='g=1')
+    plt.plot(xticks, error[:, 1, 1], marker='o', label='g=2')
+    plt.plot(xticks, error[:, 1, 2], marker='o', label='g=3')
+
+    plt.ylabel(r'$\left|\left| \frac{\phi (x) - ' +
+               r'\phi_{26}(x)}{\phi_{26}(x)} \right|\right|_2$', fontsize=14)
+    plt.yticks(fontsize=14)
+    plt.xlabel(xlabel, fontsize=14)
+    plt.xticks(xticks, fontsize=14)
+    plt.legend(loc="best", fontsize=14)
+    plt.savefig(save, dpi=300, bbox_inches="tight")
+    plt.close()
+
```

Comment:
```suggestion
    return
```

Reply:
Even better, you can (and should) return the plot object itself.